### PR TITLE
FE Loading Icon

### PR DESCRIPTION
**Pull request by peterpie123**
Adds an icon that indicates when a GET request is being executed

### CODE REVIEW

File: src/main/angular/app/graph/graph.component.ts [hunk 1]
```diff
@@ -42,22 +42,27 @@ export class GraphComponent implements OnInit {
   /** Whether to show the chart. When it's not selected, prompt the user to select a project. */
   public shouldShowChart: boolean;
 
+  /** Whether there's an active web request. */
+  public activeRequest: boolean;
```
Comment (davidgaskins) [inline src/main/angular/app/graph/graph.component.ts:46]: "hasActiveRequest"

File: src/main/angular/app/graph/graph.component.html [hunk 2]
```diff
@@ -22,7 +22,9 @@ <h1>Nothing to display! <br />
 </div>
 
 <div id="show-chart" *ngIf="shouldShowChart">
-  <div id="date-container" [style.width.px]="properties.width">
+  <div id="top-container" [style.width.px]="properties.width * 3 / 4">
+    <mat-spinner *ngIf="activeRequest" diameter="40"></mat-spinner>
```
Comment (davidgaskins) [inline src/main/angular/app/graph/graph.component.html:26]: Did you consider a progress bar instead? Without a screenshot, it's hard to tell what the better User Experience would be. 

Link to progress bar.
https://material.angular.io/components/progress-bar/overview

Comment (peterpie123) [reply src/main/angular/app/graph/graph.component.html]: Yes. To my eye, the spinner fits in better with the rest of the page. 
I think it better represents the query in progress than something like the [query progress bar](https://material.angular.io/components/progress-bar/overview#query).

**Comment by peterpie123** [comment]
OK I switched to a query loading bar from Angular and integrated it into the project-select box. That way, it's immediately clear that loading is happening when a project is selected.

**Pull request merged**